cpp/src/arrow/compute/exec.h

```diff
@@ -20,6 +20,8 @@
 
 #pragma once
 
+#include <algorithm>
```

Comment:
For the record, `algorithm` is potentially a heavy inclusion, we should perhaps avoid including it nilly-willy.